File: python/math/gcd.py
def gcd(p, q):
    """Implementation of Euclid's Algorithm to find GCD between 2 numbers"""
    if q == 0:
        return p
    return gcd(q, p % q)


def lcm(p, q):
    return (p * q)//gcd(p, q)


print(gcd(10, 5))
print(lcm(4, 6))

File: javascript/math/Gcd.js
class Gcd {
    gcd(p, q) {
        if (q === 0) return p;
        return this.gcd(q, p % q);
    }

    lcm(p, q) {
        return (p * q)/this.gcd(p, q);
    }
}

console.log(new Gcd().gcd(5,10));
console.log(new Gcd().lcm(4,6));